plugin/github/src/main/java/ca/on/oicr/gsi/shesmu/github/GitHubBranchesApiRepository.java

```diff
@@ -52,76 +86,21 @@ public void emit(SectionRenderer renderer) throws XMLStreamException {
 		}
 
 		public Stream<GithubBranchValue> stream() {
-			if (Duration.between(lastUpdated, Instant.now()).get(ChronoUnit.SECONDS) > 900) {
-				configuration.ifPresent(c -> {
-					try (AutoCloseable timer = fetchLatency.start();
-							CloseableHttpResponse response = HTTP_CLIENT
-									.execute(new HttpGet(String.format("https://api.github.com/repos/%s/%s/branches",
-											c.getOwner(), c.getRepo())))) {
-						cache = Stream
-								.of(RuntimeSupport.MAPPER.readValue(response.getEntity().getContent(),
-										BranchResponse[].class))//
-								.map(r -> new GithubBranchValue() {
-
-									@Override
-									public String branch() {
-										return r.getName();
-									}
-
-									@Override
-									public String commit() {
-										return r.getCommit().getSha();
-									}
-
-									@Override
-									public String owner() {
-										return c.getOwner();
-									}
-
-									@Override
-									public String repository() {
-										return c.getRepo();
-									}
-								})//
-								.collect(Collectors.toList());
-						count.labels(fileName().toString()).set(cache.size());
-						lastUpdated = Instant.now();
-						lastFetchTime.labels(fileName().toString()).setToCurrentTime();
-					} catch (final Exception e) {
-						e.printStackTrace();
-						provenanceError.labels(fileName().toString()).inc();
-					}
-				});
-			}
-			return cache.stream();
+			return cache.get();
 		}
 
 		@Override
 		public Optional<Integer> update(Configuration value) {
 			configuration = Optional.of(value);
+			cache.invalidate();
 			return Optional.empty();
 		}
 	}
 
-	private static final Gauge count = Gauge.build("shesmu_github_last_count", "The number of branches from GitHub.")
```

Comment:
You got rid of the monitoring in this class?

Reply:
Yes. It's handled by the caching layer now.